### PR TITLE
Fix null handling of decimal values in QueryAssertions

### PR DESCRIPTION
**Pull request by isadikov**
This PR fixes the issue of handling LongDecimal and ShortDecimal null values when comparing results in QueryAssertions.cpp. Without the fix, such comparison would fail with
```
C++ exception with description "Exception: VeloxRuntimeError
Error Source: RUNTIME
Error Code: INVALID_STATE
Reason: Use smallDecimal() or longDecimal() for DECIMAL values.
```

I also fixed QueryAssertionsTest.cpp - it was not included in any CMakeLists.txt file so it was not running at all.

### CODE REVIEW

**Comment by isadikov** [comment]
@karteekmurthys @aditi-pandit Can you review this PR?

**Comment by isadikov** [comment]
@kagamiori I made the QueryAssertionsTest runnable (it passes locally). Can you also take a look? Thanks.

**Review by karteekmurthys — APPROVED** [review]
LGTM!

**Review by aditi-pandit — APPROVED** [review]
Change looks good. Thanks for fixing.

**Review by majetideepak — APPROVED** [review]

**Comment by isadikov** [comment]
@mbasmanova Could you review this PR? Thanks.

File: velox/exec/tests/QueryAssertionsTest.cpp [hunk 1]
```diff
@@ -343,4 +343,18 @@ TEST_F(QueryAssertionsTest, multiFloatColumnWithNonUniqueKeys) {
...
+          {std::nullopt},
+          SHORT_DECIMAL(5, 2))});
+  EXPECT_TRUE(assertEqualResults({shortDecimal}, {shortDecimal}));
+
+  auto longDecimal = makeRowVector({
+      makeNullableLongDecimalFlatVector(
+          {std::nullopt},
+          LONG_DECIMAL(20, 2))});
+  EXPECT_TRUE(assertEqualResults({longDecimal}, {longDecimal}));
+}
```
Comment (kagamiori) [inline velox/exec/tests/QueryAssertionsTest.cpp:358]: For completeness, could you also add a test case where assertEqualResults fails with two decimal null vector of different precisions and scales?

**Review by kagamiori — COMMENTED** [review]
Hi @isadikov, Thank you for making the fix! It looks good to me overall. I left one comment about the test.

File: velox/exec/tests/QueryAssertionsTest.cpp (hunk 1 above)
Comment (isadikov) [reply velox/exec/tests/QueryAssertionsTest.cpp]: Done.

**Review by kagamiori — APPROVED** [review]
LGTM. Thanks!

**Comment by kagamiori** [comment]
Hi @isadikov, There are a few internal failures due to build issue. Could you try rebasing on top of the latest commit? I'll import it again then.

**Comment by isadikov** [comment]
I have rebased the branch, thanks.

**Comment by kagamiori** [comment]
Hi @isadikov, I realized that QueryAssertionsTest.cpp was moved from velox/exec/tests/utils to velox/exec/tests and this caused the internal build issues. Could it be moved back?

**Comment by isadikov** [comment]
I moved the file to be grouped together with the rest of the test files - this is where CMakeLists.txt defines the test target to run all of the tests in the directory and it was easier to add the test.

**Comment by isadikov** [comment]
I can see if it is possible to keep in the original directory if you like. Could you share the error, maybe I can fix the error instead?

**Comment by kagamiori** [comment]
> I can check if it is possible to keep in the original directory if you like. Could you share the error, maybe I can fix the error instead?

The errors are about our internal build files. (We don't use CMakeList files but a separate files not included in github.) Is there a reason for this move? If not, let's keep it at the original place.

**Comment by kagamiori** [comment]
> I moved the file to be grouped together with the rest of the test files - this is where CMakeLists.txt defines the test target to run all of the tests in the directory and it was easier to add the test.

Sounds good. Then let's keep the move and I'll fix the internal build errors.

**Comment by isadikov** [comment]
Okay, thanks! Let me know if any changes required from me.

**Comment by isadikov** [comment]
@kagamiori Is there anything needed from my end to help with the PR merge?
